packages/core/utils/src/common/errors.ts

```diff
@@ -46,6 +46,8 @@ export class MedusaError extends Error {
   constructor(type: string, message: string, code?: string, ...params: any) {
     super(...params)
 
+    this.name = this.constructor.name
```

Comment:
I would suggest to do the following to follow our practices and be consistent with the rest of the code base

```ts
export class MedusaError extends Error {
  // we do not use Symbol because it is not serializable
  __isMedusaError = true

  public type: string
  public message: string
  public code?: string
  public date: Date
  public static Types = MedusaErrorTypes
  public static Codes = MedusaErrorCodes

  static isMedusaError(error: any): error is MedusaError {
    return !!error.__isMedusaError
  }
```

and then when checking somewhere you can do `MedusaError.isMedusaError(err)`

wdyt?

Reply:
You have your point :+1: 

In that case, I would add a hint in the docs as well?

Reply:
Sure, open that as a different PR as it requires different reviewers.